src/apm.js

```diff
@@ -19,21 +19,35 @@
 
 const { existsSync } = require('fs');
 const { join } = require('path');
-const { name, version } = require('../package.json');
+const { name, version, build } = require('../package.json');
 
-module.exports = function (serviceName = name) {
-  if (process.env.kbnWorkerType === 'optmzr') return;
+/**
+ * flag to disable APM support on all kibana builds by default
+ */
+const isKibanaDistributable = Boolean(build && build.distributable === true);
 
-  const conf = {
+function getConfig(serviceName) {
+  const config = {
     serviceName: `${serviceName}-${version.replace(/\./g, '_')}`
   };
 
-  if (configFileExists()) conf.configFile = 'config/apm.js';
-  else conf.active = false;
+  const configPath = join(__dirname, '..', 'config', 'apm.js');
+  if (existsSync(configPath)) {
+    const configFile = require(configPath); // eslint-disable-line import/no-dynamic-require
+    Object.assign(config, configFile);
+  } else {
+    config.active = false;
```

Comment:
When manually merging the `configFile` options, there's no longer any reason to have this `else` block. Just move the `active: false` up to where the `config` object is created.